protob/messages-ontology.proto

```diff
@@ -0,0 +1,147 @@
+    repeated uint32 address_n = 1;  // BIP-32 path to derive the key from master node
+    optional bool show_display = 2; // Optionally show on display before sending the result
+}
+
+/**
+ * Response: Contains Ontology public key derived from device private seed
+ * @prev OntologyGetPublicKey
+*/
+message OntologyPublicKey {
+    optional bytes public_key = 1;  // Ontology public key
+}
+
+/**
+ * Request: Ask device for Ontology address corresponding to address_n path
+ * @next OntologyAddress
+ */
+message OntologyGetAddress {
+    repeated uint32 address_n = 1;  // BIP-32 path to derive the key from master node
+    optional bool show_display = 2; // Optionally show on display before sending the result
+}
+
+/**
+ * Response: Contains Ontology address derived from device private seed
+ * @prev OntologyGetAddress
+ */
+message OntologyAddress {
+    optional string address = 1;    // Ontology address
+}
+
+
+/**
+ * Request: Ask device to sign Ontology transfer
+ * @next OntologySignedTransfer
+ */
+ message OntologySignTransfer {
+    repeated uint32 address_n = 1;                  // BIP-32 path to derive the key from master node
+    optional OntologyTransaction transaction = 2;
+    optional OntologyTransfer transfer = 3;
+    optional bool return_payload = 4;
+    
+    /**
+    * Ontoloy Transfer
+    */
+    message OntologyTransfer {
+        /**
+        * Ontoloy Asset
+        */
+        enum OntologyAsset {
+            ONT = 1;
+            ONG = 2;
+        }
+
+        optional OntologyAsset asset = 1;
+        optional uint64 amount = 2;
+        optional string fromAddress = 3;
+        optional string toAddress = 4;
+    }
+}
+
+/**
+* Response: Contains Ontology transfer signature
+* @prev OntologySignTransfer
+*/
+message OntologySignedTransfer {
+    optional bytes signature = 1;
+    optional bytes payload = 2;
+}
+
+/**
+ * Request: Ask device to sign Ontology ONG withdrawal
+ * @next OntologySignedWithdrawOng
+ */
+ message OntologySignWithdrawOng {
+    repeated uint32 address_n = 1;                  // BIP-32 path to derive the key from master node
+    optional OntologyTransaction transaction = 2;
+    optional OntologyWithdrawOng withdrawOng = 3;
+    optional bool return_payload = 4;
+    
+    /**
+    * Ontoloy ONG Withdrawal
+    */
+    message OntologyWithdrawOng {
+        optional uint64 amount = 3;
+        optional string fromAddress = 4;
+        optional string toAddress = 5;
+    }
+}
+
+/**
+* Response: Contains Ontology ONG withdrawal signature
+* @prev OntologySignWithdrawOng
+*/
+message OntologySignedWithdrawOng {
+    optional bytes signature = 1;
+    optional bytes payload = 2;
+}
+
+/**
+ * Request: Ask device to sign Ontology arbitrary transaction
+ * @next OntologySignedTx
+ */
+ message OntologySignTx {
+    repeated uint32 address_n = 1;                  // BIP-32 path to derive the key from master node
+    optional OntologyTransaction transaction = 2;
+    optional bytes payload = 3;
```

Comment:
We don't allow signing unknown scripts. I suggest to drop this message completely.

Reply:
That's reasonable. There a few more types of transaction I need to add besides transfer and withdrawOng. Mainly registering Identity on Ontology network (that's no problem) and calling arbitrary smart contract. 

Will you accept signing such call? The script will be build from Smart contract address, smart contract method and arbitrary parameters. I can not guaranty what will be the outcome of calling such contract.

Reply:
Let's create a message for registering an Identity.

I am not so sure about SmartContractCall message, because we cannot show arbitrary parameters in a meaningful way on Trezor. So I would leave that out for the time being.